acceptance-tests/dsl/src/main/java/org/hyperledger/besu/tests/acceptance/dsl/node/configuration/BesuNodeFactory.java

```diff
@@ -499,14 +499,24 @@ public BesuNode createQbftNodeWithTLSPKCS11(final String name) throws IOExceptio
   }
 
   public BesuNode createQbftNode(final String name) throws IOException {
+    JsonRpcConfiguration rpcConfig = node.createJsonRpcWithQbftEnabledConfig(false);
+    rpcConfig.setPort(
+        Math.abs(name.hashCode() % 60000)
+            + 1024); // Generate a consistent port for p2p based on node name
+    rpcConfig.addRpcApi("ADMIN,TXPOOL");
     return create(
         new BesuNodeConfigurationBuilder()
             .name(name)
             .miningEnabled()
-            .jsonRpcConfiguration(node.createJsonRpcWithQbftEnabledConfig(false))
+            .jsonRpcConfiguration(rpcConfig)
             .webSocketConfiguration(node.createWebSocketEnabledConfig())
             .devMode(false)
             .genesisConfigProvider(GenesisConfigurationFactory::createQbftGenesisConfig)
+            .p2pPort(
+                Math.abs(name.hashCode() % 60000)
+                    + 1024
+                    + 500) // Generate a consistent port for p2p based on node name (+ 500 to avoid
+            // clashing with RPC port or other nodes with a similar name)
```

Comment:
fixing the ports makes me nervous - we've had flaky tests for this reason in the past. Realise it's unlikely but will it be obvious if there is a port conflict at this point?

Reply:
createQbftNode is used by other tests so would not want to introduce any flakiness here even if it's unlikely

Reply:
@matthew1001 Would be a good idea to refactor method with signature `public BesuNode createQbftNode(final String name, final int port)`. Then the existing method can pass `0` to use random port? while new tests can pass calculated ports?

Reply:
Yes I'd be happy to do that if we think fixing ports in the way I have done is going to be problematic. I'd personally vote for it being fairly stable (certainly the existing acceptance tests all still pass) but v happy to refactor in this way if the consensus is to do so.